H5P.BranchedVideo/branched-video.js

```diff
@@ -0,0 +1,847 @@
+        videoDiv.style.display = 'block';
+        var video = this.getVideoHTML();
+        video.play();
+      }
+      // sets type of videos
+      this.setType = function(text){
+        this.type = text;
+      }
+      // gets type of Videos
+      this.getType = function(){
+        return this.type;
+      }
+
+    } // end of branch
+
+
+
+    // NODE
+    function Node(subBranch, parentSlug){
+      if (subBranch.branchSlug == undefined){return null;}
+      this.startTime = subBranch.branchTimeFrom;
+      this.endTime = subBranch.branchTimeTo;
+      this.branchSlug = subBranch.branchSlug;
+      this.link =  new Link(subBranch.bubble, this.branchSlug);
+      this.getNodeHTML = function(){
+        return this.link.getLinkHTML();
+      }
+      // functions to show and hide
+      this.showLink = function(){
+        var linkHTML = this.getNodeHTML();
+        linkHTML.style.display='block';
+      }
+      this.hideLink = function(){
+        var linkHTML = this.getNodeHTML();
+        linkHTML.style.display='none';
+      }
+      // BUBBLE
+      function Link(bubble, slug){
+        this.text = bubble.text;
+        this.style = bubble.style;
+        this.color = bubble.color;
+        this.positionX = bubble.positionX;
+        this.positionY = bubble.positionY;
+      // returns the HTML component
+        this.createLinkHTML = function(){
+          if (slug == undefined){
+            return null;
+          }
+          var currentLink = document.createElement('button');
+          currentLink.id = 'tapestry-link-' + slug;
+          currentLink.className = 'tapestry-branch-button';
+          currentLink.type = 'button';
+          currentLink.innerHTML = this.text;
+          currentLink.style.position = 'absolute';
+          currentLink.style.left = this.positionX + '%';
+          currentLink.style.top = this.positionY + '%';
+          currentLink.style.zIndex = 2147483647;
+          currentLink.style.display = 'none';
+          return currentLink;
+        }
+        this.linkHTML = this.createLinkHTML();
+        this.getLinkHTML = function(){
+          return this.linkHTML;
+        }
+      }// end of bubble
+    }// end of node
+
+
+
+
+//  INITIALIZE
+    var helpMode = false;
+    var mainSlug = this.options.mainBranchSlug;
+    var currentVideoPlaying = mainSlug;
+    // sets background to black
+    $container.css({'background-color':'#000001'});
+    // create associative array temporarily
+    var branched_videos = {};
+    // given slug, returns branch
+    function getBranch(slug){
+      return branched_videos[slug];
+    }
+    // given slug, creates time text
+    function createTimeText(slug){
+      var branch = getBranch(slug);
+      var lengthTime = branch.videoLength;
+      var mins = Math.floor(lengthTime / 60);
+      var seconds = Math.floor(lengthTime % 60);
+      if (seconds < 10){
+        seconds = "0"+ seconds;
+      }
+      var para = document.createElement('p');
```

Comment:
Could you please name your variables a little bit better so it's easy to know exactly what they are? If this is a label to show time, maybe you could name it `timeLabel` (follow the same logic for the rest of your code)